### ISSUE TITLE
c-ares ignores domain search suffix on Android

### ISSUE THREAD

**Issue by ArtsviKarayan** [c-ares/c-ares#207]
On Linux the domain search is retrieved from the resolve.conf.
The following example does not work on Android (e.g. ok on Linux):
- Domain search: "example.domain.search.com"
- Actual domain: "actualdom.example.domain.search.com"
- Name passed to ares_gethostbyname: "actualdom"
- Result: No resolved ip

I have investigated the sources little bit and it seems that the "domain search" config is not initialized on Android. The name servers are retrieved from the Android system properties but the domain search property is not retrieved and used any more.

**Comment by bradh352** [comment]
You are right, it would make sense to also pull in getDomains() from the link properties. 
 Think this was just skipped due to the sheer complexity of calling Java from C, but it would be good to add.  Pulling in the original submitter of the modern Android code for c-ares, @user-none

https://developer.android.com/reference/android/net/LinkProperties.html#getDomains()

**Comment by user-none** [comment]
Pull request for this #211.
